### PR TITLE
add grafana views for tekton chains controller workque and latency

### PR DESCRIPTION
**Pull request by jkhelil**
A followup to https://github.com/openshift-pipelines/pipeline-service/pull/989(closed accidently, and not able to reopen it)

### CODE REVIEW

**Comment by gabemontero** [comment]
yeah I've never seen that being able to reopen a PR before @jkhelil  .... the same thing shows up for me.  No idea what is going on.

in any event, yes, the screen shot at https://github.com/openshift-pipelines/pipeline-service/pull/989#issuecomment-2074288670 was along the lines of what I was talking about.  Thanks.

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 1]
```diff
@@ -3892,6 +4100,10 @@
           ],
           "title": "Tekton Chains CPU Fraction Use",
           "type": "timeseries"
+        }
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:4103]: any idea why this `Tekton Chains CPU Fraction Use` was repeated @jkhelil  .... perhaps the title is wrong and it needs to changes to reflect what the new panel is ?

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 2]
```diff
@@ -3810,6 +3810,129 @@
       },
       "id": 79,
       "panels": [
+        {
+          "description": "Watcher work queue depth",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:3814]:
```suggestion
          "description": "Tekton Chains work queue depth",
```

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 3]
```diff
@@ -3810,6 +3810,129 @@
...
+              "exemplar": true,
+              "expr": "sum(watcher_workqueue_depth{container='tekton-chains-controller',app='tekton-chains-controller'})",
+              "interval": "",
+              "intervalFactor": 1,
+              "legendFormat": "Queue Depth",
+              "range": true,
+              "refId": "work queue depth"
+            }
+          ],
+          "title": "Watcher Work Queue Depth",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:3871]:
```suggestion
          "title": "Tekton Chains Work Queue Depth",
```

**Review by gabemontero — CHANGES_REQUESTED** [review]
the queries look OK, but some of the titles etc. that you copied over from the results panel were not updated to say tekton chains @jkhelil

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 4]
```diff
@@ -3810,6 +3810,129 @@
...
+              "exemplar": true,
+              "expr": "histogram_quantile(0.99, sum(rate(watcher_reconcile_latency_bucket{job=\"tekton-chains\"}[30m])) by (le) ) / 1000",
+              "interval": "",
+              "intervalFactor": 4,
+              "legendFormat": "Latency",
+              "range": true,
+              "refId": "reconcile latency"
+            }
+          ],
+          "title": "Watcher Reconcile Latency",
```
[the remark this answers is not in the document]
Comment (jkhelil) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json]: fixed hereis a final screenshot 
image

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json (hunk 1 above)
Comment (jkhelil) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:4103]: duplicate, removed now

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 5]
```diff
@@ -3810,6 +3810,129 @@
...
+            },
+            "text": {},
+            "textMode": "auto"
+          },
+          "pluginVersion": "9.5.17",
+          "targets": [
+            {
+              "editorMode": "code",
+              "exemplar": true,
+              "expr": "sum(watcher_workqueue_depth{container='tekton-chains-controller',app='tekton-chains-controller'})",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:3863]: I validated this query works in prod-rh01 from the OCP metrics console.

In theory you should have permissions to do that as well @jkhelil 

If you get a sec, try logging into the OCP console using the red hat SSO and try your query over at https://console-openshift-console.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/monitoring/query-browser?query0=

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json (hunk 2 above)
Comment (gabemontero) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:3814]: You still have to this one to update @jkhelil

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 6]
```diff
@@ -3810,6 +3810,129 @@
...
+            },
+            "text": {},
+            "textMode": "auto"
+          },
+          "pluginVersion": "9.5.17",
+          "targets": [
+            {
+              "editorMode": "code",
+              "exemplar": true,
+              "expr": "histogram_quantile(0.99, sum(rate(watcher_reconcile_latency_bucket{job=\"tekton-chains\"}[30m])) by (le) ) / 1000",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:3925]: I had to convert `\"` to just `"` @jkhelil but `histogram_quantile(0.99, sum(rate(watcher_reconcile_latency_bucket{job="tekton-chains"}[30m])) by (le) ) / 1000` also works from the OCP metrics console ... see if you can do it as well

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 7]
```diff
@@ -3884,13 +4008,13 @@
           "targets": [
             {
               "editorMode": "builder",
-              "expr": "rate(watcher_go_gc_cpu_fraction{app=\"tekton-chains-controller\"}[5m])",
+              "expr": "rate(watcher_workqueue_longest_running_processor_seconds_count{app=\"tekton-chains-controller\"}[5m])",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:4011]: Not that I'm necessarily opposed to this change, but this was not one we discussed @jkhelil  ... explain to me why you went in this direction

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json [hunk 8]
```diff
@@ -3967,13 +4092,13 @@
           "targets": [
             {
               "editorMode": "builder",
-              "expr": "rate(watcher_workqueue_longest_running_processor_seconds_count{app=\"tekton-chains-controller\"}[5m])",
+              "expr": "rate(watcher_go_gc_cpu_fraction{app=\"tekton-chains-controller\"}[5m])",
```
Comment (gabemontero) [inline operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:4095]: Not that I'm necessarily opposed to this change, but this was not one we discussed @jkhelil  ... explain to me why you went in this direction

**Review by gabemontero — CHANGES_REQUESTED** [review]
closer but a couple of more items @jkhelil 

also, I suggest you try some of the queried from Konflux prod or stage if you have not already for practice

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json (hunk 2 above)
Comment (jkhelil) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json]: fixed

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json (hunk 6 above)
Comment (jkhelil) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json:3925]: it is just a way to escape doublequote inside the doublequote, it is used all over the json file

File: operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json (hunk 7 above)
Comment (jkhelil) [reply operator/gitops/argocd/grafana/dashboards/pipeline-service-dashboard.json]: there was no change,just inversed the order for the two metrics, it is now ok

**Comment by gabemontero** [comment]
looks good diff wise @jkhelil 
also checked out your PR and tried it locally
merging

**Pull request merged**